core/src/Streamly/Internal/Unicode/Stream.hs

```diff
@@ -1076,8 +1073,10 @@ stripHead = S.dropWhile isSpace
 --
 -- /Pre-release/
 {-# INLINE lines #-}
-lines :: (Monad m, IsStream t) => Fold m Char b -> t m Char -> t m b
-lines = S.splitOnSuffix (== '\n')
+lines :: (Monad m) => Fold m Char b -> Stream.Stream m Char -> Stream.Stream m b
+lines f m =
+    Stream.fromStreamD $
+    D.foldMany (Fold.takeEndBy_ (== '\n') f) (Stream.toStreamD m)
```

Comment:
Indent this line.

Instead of using D.foldMany we can use Stream.foldMany, that will avoid using StreamD.
Need to check the performance, if there is degradation we can use D.splitOnSuffix here.